src/test/scala/uk/gov/hmrc/play/audit/http/HttpAuditingSpec.scala

```diff
@@ -213,6 +214,108 @@ class HttpAuditingSpec extends WordSpecLike with Matchers with Inspectors with E
       dataEvent.response.generatedAt shouldBe responseDateTime
 
     }
+
+    "mask passwords in an OutboundCall using form values" in {
+      val connector = mock[AuditConnector]
+      val httpWithAudit = new HttpWithAuditing(connector)
+
+      val requestBody = Map("ok" -> "a", "password" -> "hide-me")
+      val response = new DummyHttpResponse(Json.obj("password" -> "hide-me").toString, 200)
+
+      val request = httpWithAudit.buildRequest(serviceUri, "POST", Some(requestBody))
+
+      whenAuditSuccess(connector)
+
+      httpWithAudit.audit(request, response)
+
+      val dataEvent = verifyAndRetrieveEvent(connector)
+
+      dataEvent.request.detail(RequestBody) shouldBe requestBody.toString.replaceAllLiterally("hide-me", "########")
+      dataEvent.response.detail(ResponseMessage) shouldBe response.body.replaceAllLiterally("hide-me", "########")
+    }
+
+    "mask passwords in an OutboundCall using json" in {
+      val connector = mock[AuditConnector]
+      val httpWithAudit = new HttpWithAuditing(connector)
+
+      val sampleJson = Json.obj("a" -> Json.arr(Json.obj("ok" -> 1, "password" -> "hide-me", "PASSWD" -> "hide-me")))
+      val response = new DummyHttpResponse(Json.stringify(sampleJson), 200)
+
+      val request = httpWithAudit.buildRequest(serviceUri, "POST", Some(Json.stringify(sampleJson)))
+
+      whenAuditSuccess(connector)
+
+      httpWithAudit.audit(request, response)
+
+      val dataEvent = verifyAndRetrieveEvent(connector)
+
+      dataEvent.request.detail(RequestBody) shouldBe sampleJson.toString.replaceAllLiterally("hide-me", "########")
+      dataEvent.response.detail(ResponseMessage) shouldBe response.body.replaceAllLiterally("hide-me", "########")
+    }
+
+    "mask passwords in an OutboundCall using xml" in {
+      val connector = mock[AuditConnector]
+      val httpWithAudit = new HttpWithAuditing(connector)
+
+      val sampleXml =
+        """<abc>
+          |    <foo>
+          |        <Password>hide-me</Password>
+          |        <prefix:password>hide-me</prefix:password>
```

Comment:
One of the tests could use sth like `userPassword` to confirm the `contains` bit of checking. It may be that `prefix:password` already does it